plugins/outputs/stomp/stomp.go

```diff
@@ -0,0 +1,114 @@
+package stomp
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+
+	"github.com/go-stomp/stomp"
+
+	"github.com/influxdata/telegraf"
+	commontls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+type STOMP struct {
+	Host      string          `toml:"host"`
+	Username  string          `toml:"username,omitempty"`
+	Password  string          `toml:"password,omitempty"`
+	QueueName string          `toml:"queueName"`
+	Log       telegraf.Logger `toml:"-"`
+
+	commontls.ClientConfig
+
+	conn  net.Conn
+	stomp *stomp.Conn
+
+	serialize serializers.Serializer
+}
+
+func (q *STOMP) Connect() error {
+	tlsConfig, err := q.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+	if tlsConfig != nil {
+		q.conn, err = tls.Dial("tcp", q.Host, tlsConfig)
+		if err != nil {
+			return err
+		}
+	} else {
+		q.conn, err = net.Dial("tcp", q.Host)
+		if err != nil {
+			return err
+		}
+	}
+
+	q.stomp, err = stomp.Connect(q.conn, stomp.ConnOpt.HeartBeat(0, 0), stomp.ConnOpt.Login(q.Username, q.Password))
+	if err != nil {
+		return err
+	}
+	q.Log.Info("STOMP Connected...")
+	return nil
+}
+
+//SetSerializer ...
+func (q *STOMP) SetSerializer(serializer serializers.Serializer) {
+	q.serialize = serializer
+}
+
+//Write ...
+func (q *STOMP) Write(metrics []telegraf.Metric) error {
+	for _, metric := range metrics {
+		values, err := q.serialize.Serialize(metric)
+		if err != nil {
+			q.Log.Errorf("Serializing metric %v failed: %s", metric, err)
+			continue
+		}
+		err = q.stomp.Send(q.QueueName, "text/plain",
+			[]byte(values), nil)
+		if err != nil {
+			return fmt.Errorf("sending metric failed: %s", err)
+		}
+	}
+	return nil
+}
+
+//Close ...
+func (q *STOMP) Close() error {
+	return q.stomp.Disconnect()
+}
+
+//SampleConfig ...
+func (q *STOMP) SampleConfig() string {
+	return `
+	## Host of Active Mq broker
+	host = "localhost:61613"
+	## Queue name for producer messages
+	queueName = "telegraf"
+	## Optional credentials
+	# username = ""
+	# password = ""
+	## Optional TLS Config (defaults to unencrypted connection)
+	# tls_ca = "/etc/telegraf/ca.pem"
+	# tls_cert = "/etc/telegraf/cert.pem"
+	# tls_key = "/etc/telegraf/key.pem"
+   
+	## Use TLS but skip chain & host verification
+	# insecure_skip_verify = false
+	## Data format to output.
+	## Each data format has it's own unique set of configuration options, read
+	## more about them here:
+	## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+	# data_format = "json"
+`
+}
```

Comment:
The sample-config needs to start with two-spaces, therefore most plugins declare a `sampleConfig` constant at the root level to make this nice. However, you can also write
```suggestion
func (q *STOMP) SampleConfig() string {
	return `
  ## Host of Active Mq broker
  host = "localhost:61613"
  ## Queue name for producer messages
  queueName = "telegraf"
  ## Optional credentials
  # username = ""
  # password = ""
  ## Optional TLS Config (defaults to unencrypted connection)
  # tls_ca = "/etc/telegraf/ca.pem"
  # tls_cert = "/etc/telegraf/cert.pem"
  # tls_key = "/etc/telegraf/key.pem"
   
  ## Use TLS but skip chain & host verification
  # insecure_skip_verify = false
  ## Data format to output.
  ## Each data format has it's own unique set of configuration options, read
  ## more about them here:
  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
  # data_format = "json"
`
}
```

Please also synchronize the settings with the README, e.g. spaces between "sections" and comment of the `queueName` option...